src/Magnum/Audio/Source.cpp

```diff
@@ -38,6 +41,32 @@ Source& Source::setBuffer(Buffer* buffer) {
     return *this;
 }
 
+Source& Source::queueBuffers(Containers::ArrayView<Containers::Reference<Buffer>> buffers) {
+    Containers::Array<ALuint> ids(buffers.size());
+    for(std::size_t i = 0; i < ids.size(); i++)
+        ids[i] = buffers[i]->id();
+    alSourceQueueBuffers(_id, ids.size(), ids.data());
+    return *this;
+}
+
+std::size_t Source::unqueueBuffers(Containers::ArrayView<Containers::Reference<Buffer>> buffers) {
+    ALint processedBuffers;
+    alGetSourcei(_id, AL_BUFFERS_PROCESSED, &processedBuffers);
+
+    if(!processedBuffers) return 0;
+
+    Containers::Array<ALuint> unqueuedIds(processedBuffers);
+    alSourceUnqueueBuffers(_id, unqueuedIds.size(), unqueuedIds.data());
+    auto isNotUnqueued = [&unqueuedIds](Containers::Reference<Buffer> buffer){
```

Comment:
would the following work too?

```suggestion
    auto isNotUnqueued = [&unqueuedIds](Buffer& buffer) {
```

Reply:
I forgot that `Reference<Type>` was convertible to `Type&`. 🤦‍♂